### PR TITLE
Update pom.xml 

### PR DESCRIPTION
**Pull request by markro49**
update several plugins to latest version
update CheckerFramework version
create two profiles so same pom works for both JDK 8 and JDK 11

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,13 +27,13 @@
   <parent>
     <groupId>org.apache.commons</groupId>
     <artifactId>commons-parent</artifactId>
-    <version>49</version>
+    <version>50</version>
```
Comment (mernst) [inline pom.xml:30]: This change doesn't seem related to the pull request description.  Should it be undone?

File: pom.xml [hunk 2]
```diff
@@ -27,13 +27,13 @@
...
     <artifactId>commons-parent</artifactId>
-    <version>49</version>
+    <version>50</version>
   </parent>
 
   <groupId>org.apache.bcel</groupId>
   <artifactId>bcel</artifactId>
   <packaging>jar</packaging>
-  <version>6.4.1</version>
+  <version>6.4.1.1</version>
```
Comment (mernst) [inline pom.xml:36]: This change is not related to the pull request description.

File: pom.xml [hunk 3]
```diff
@@ -492,9 +486,6 @@
         <version>${commons.rat.version}</version>
         <configuration>
           <excludes>
-	    <exclude>README-typetools.txt</exclude>
```
Comment (mernst) [inline pom.xml:495]: Why are these exclusions removed?

File: pom.xml [hunk 4]
```diff
@@ -42,20 +42,17 @@
...
-
+    <checkerFrameworkVersion>3.2.0</checkerFrameworkVersion>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
     <commons.componentid>bcel</commons.componentid>
     <commons.module.name>org.apache.bcel</commons.module.name>
-    <commons.release.version>6.4.1</commons.release.version>
+    <commons.release.version>6.4.1.1</commons.release.version>
```
Comment (mernst) [inline pom.xml:52]: I'm not sure why this change is in this pull request.

**Review by mernst — CHANGES_REQUESTED** [review]
There are a few changes that look suspicious to me.
As a general rule, it is better to make a separate pull request for each change:
 * updating the Checker Framework version
 * updating the maven-compiler-plugin version
Those are easier to understand both now and when someone looks over the version history.

File: src/main/java/org/apache/bcel/util/package.html [hunk 5]
```diff
@@ -17,17 +17,13 @@
...
-$Id$
--->
 </head>
 <body bgcolor="white">
 <p>
 This package contains utility classes for the
 <a href="https://commons.apache.org/bcel/">Byte Code Engineering
 Library</a>, namely:
 </p>
-<p>
```
Comment (mernst) [inline src/main/java/org/apache/bcel/util/package.html:30]: Rather than putting these differences in this pull request, it would be nicer to:
 * pull apache fork into codespecs fork
 * pull codespecs fork into typetools fork
Now, this change is not needed, and as a side benefit the forks are more up to date.

Comment (markro49) [reply src/main/java/org/apache/bcel/util/package.html:30]: done
